module.exports = (fn) => {
  return (req, res, next) => {
    fn(req, res, next).catch((err) => {
      next(err); // âœ… Always pass errors to next()
    });
  };
};
